src/main/java/org/broadinstitute/hellbender/tools/walkers/readorientation/CollectF1R2Counts.java

```diff
@@ -0,0 +1,233 @@
+        if (referenceSite) {
+            refSiteHistograms.get(refContext).increment(depth <= F1R2FilterConstants.maxDepth ?
+                    depth : F1R2FilterConstants.maxDepth);
+            return;
+        }
+
+        // If we got here, we have an alt site with a single alt base
+        final Nucleotide altBase = Nucleotide.valueOf(BaseUtils.baseIndexToSimpleBase(altBaseIndex));
+
+        final int refCount = baseCounts[refBase.ordinal()];
+        final int altCount = baseCounts[altBaseIndex];
+        Utils.validate(altCount > 0, "We must have a nonzero alt read but got " + altCount);
+
+        final int refF1R2 = pileup.getNumberOfElements(pe -> Nucleotide.valueOf(pe.getBase()) == refBase && ReadUtils.isF1R2(pe.getRead()));
+        final int altF1R2 = pileup.getNumberOfElements(pe -> Nucleotide.valueOf(pe.getBase()) == altBase && ReadUtils.isF1R2(pe.getRead()));
+
+        // If there's only one alt read at depth > minimumDepthForOptimization, we store the data differently to save space
+        if (altCount == 1) {
+            final ReadOrientation type = altF1R2 == 1 ? F1R2 : F2R1;
+            depthOneAltHistograms.get(refContext).get(new Pair<>(altBase, type))
+                    .increment(depth <= F1R2FilterConstants.maxDepth ? depth : F1R2FilterConstants.maxDepth);
+            return;
+        }
+
+        try {
+            altTableWriter.writeRecord(new AltSiteRecord(refContext, refCount, altCount, refF1R2, altF1R2, altBase));
+        } catch (IOException e) {
+            throw new UserException("Encountered an IO Exception writing to the alt data table", e);
+        }
+    }
+
+    @Override
+    public Object onTraversalSuccess() {
+        refSiteHistograms.values().forEach(h -> refMetricsFile.addHistogram(h));
+        refMetricsFile.write(refMetricsOutput);
+
+        depthOneAltHistograms.values().forEach(hs -> hs.values().forEach(h -> altMetricsFile.addHistogram(h)));
+        altMetricsFile.write(altMetricsOutput);
+
+        return "SUCCESS";
+    }
+
+    @Override
+    public void closeTool() {
+        if (altTableWriter != null) {
+            try {
+                altTableWriter.close();
+            } catch (IOException e) {
+                throw new UserException("Encountered an IO exception while closing the alt table writer", e);
+            }
+        }
+    }
+
+    /**
+     * Use a series of heuristics to detect a bad pileup.
+     */
+    private boolean isPileupGood(final ReadPileup pileup, final ReferenceContext referenceContext){
+        // This case should not happen, as AlignmentContext should come filtered, but it does happen once in a while
+
+        final int[] baseCounts = pileup.getBaseCounts();
+        final int depth = (int) MathUtils.sum(baseCounts);
+
+        // If observe any bases that are neither ref nor alt, the site is no good so filter
+        final long numAlleles = Arrays.stream(baseCounts).filter(i -> i > 0).count();
```

Comment:
This will break down at very high depth.  Maybe allow for other alts at a fraction of 1 in 500 or something?

Reply:
I'm thinking of just removing this criterion altogether. If there're more than one alt allele, just use the alt with highest count and ignore the other. What do you think?

Reply:
I removed this criterion. This will potentially weaken the signal for artifact if a variant and an orientation artifact strike at the same site, but this should be rare and there should be plenty of evidence for artifact elsewhere.

Reply:
That sounds fine.